src/main/java/com/univocity/parsers/containers/Row.java

```diff
@@ -0,0 +1,66 @@
+package com.univocity.parsers.containers;
+
+import java.util.Arrays;
+
+/**
+ * @author naveen.kasthuri
+ * A convenient container of information represented in a csv row. Instead of handling lists of
+ * arrays of strings, a csv can be fully represented by a list of rows.
+ *
+ * The row object acts as a map from header to a cell. For example: row.get("Make") will return the
+ * value of this row's cell under column "Make". However, each row object does not duplicate the
+ * keys (headers) like a typical Map would: only one copy of the keys is stored in the {@link Csv}
+ * object. As such, this is a memory efficient, computationally efficient, convenient container of
+ * row objects.
+ *
+ * @see com.univocity.parsers.examples's RowObjectExample for an implementation example
+ */
+public class Row {
+  private final String[] values;
+  private final int lineNumber;
+  private final Csv csv;
+
+  public Row(String[] row, int lineNumber, Csv csv) {
+    this.lineNumber = lineNumber;
+    this.values = row.clone();
+    this.csv = csv;
+  }
+
+  public int getLineNumber() {
+    return lineNumber;
+  }
+
+  /**
+   * Returns the value of this row for the given column header.
+   * @param column String that matches csv's column header.
+   * @return null if no such header is present in csv.
+   */
+  public String get(String column) {
+    if (csv.getIndex(column) != null) {
+      return values[csv.getIndex(column)];
+    }
+    return null;
+  }
+
+  /**
+   * Returns the value of this row for the given {@link Column}.
+   * @param column {@link Column}
+   * @return null if no such header is present in csv.
+   */
+  public String get(Column column) {
```

Comment:
You might also want to add convenience methods to return typed values here:

```
    T get(Column column, Class<T> expectedType){
        String stringValue = get(column);
        Conversion<String, T> someConversion = getConversionToType(expectedType); // need to implement this
        if(someConversion == null){
              throw new IllegalArgumentException("Unsupported conversion of value ["+stringValue+"] at column "+column+" to type " + expectedType.getName());        
        }
        return someConversion.convert(stringValue)
    }
```

Have a look at the classes that implement `Conversion`. You can probably cache the expected type and the column name to reuse a particular conversion implementation for the column.